gemrb/core/Scriptable/Actor.cpp

```diff
@@ -4246,7 +4246,7 @@ bool Actor::CheckSpellDisruption(int damage, int spellLevel)
 	int concentration = GetSkill(IE_CONCENTRATION);
 	int bonus = 0;
 	// combat casting bonus only applies when injured
-	if (HasFeat(FEAT_COMBAT_CASTING) && BaseStats[IE_HITPOINTS] != Modified[IE_HITPOINTS]) {
+	if (HasFeat(FEAT_COMBAT_CASTING) && Modified[IE_MAXHITPOINTS] != Modified[IE_HITPOINTS]) {
```

Comment:
probably too strong either way, but we haven't reverse engineered this — it could be it's applied only when an injured hp interval or worse is in effect.

Good catch.

Reply:
Again, I've no idea how it's supposed to behave as I don't even have IWD2.  I just saw HP being compared with itself and guessed the intended meaning.